### PR TITLE
replace math/rand with the rand pkg from k8s.io/apimachinery

### PR DESCRIPTION
**Pull request by tariq1890**
Fixes the `gosec G404` linting issues
Removed the need to add exclude directives to the golangci-lint config file

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 run:
-  deadline: 10m
+  timeout: 10m
```
Comment (elezar) [inline .golangci.yml:2]: This change seems reasonable.

**Review by elezar — COMMENTED** [review]
As was the case with the changes for https://github.com/NVIDIA/k8s-device-plugin/pull/629 I don't think these are required. These are in tests and we KNOW that the use of `math/rand` will not cause issues in this case.

Although the use of `k8s.io/apimachinery/pkg/util/rand` addresses the issues, it makes it seem as if the tests actually requrie this package -- instead of using the standard library -- and increases our external dependencies.

**Comment by klueska** [comment]
I don't feel strongly either way, but I do agree with the sentiment of not pulling in dependencies unless absolutely necessary. Could we move to `crypto/rand` instead?

**Comment by elezar** [comment]
If I recall correctly, switching to `crypto/rand` required a number of additional code changes and since these are `_tests.go` files and we're just using `rand` to generate a random integer:
```
			successAfter := rand.Intn(uniquePermutations(mc)) + 1
```
these changes are not required.

File: .golangci.yml (hunk 1 above)
Comment (tariq1890) [reply .golangci.yml:2]: I'll file another PR for it

**Pull request closed without merging**